### ISSUE TITLE
#246 - Discarding empty descriptions

### ISSUE THREAD

**Issue by fatmau** [OpenConceptLab/oclapi#341]
according to comments at #246, this story is generated.

**Acceptance Criteria:**
At Basic and Open MRS validation schema;

- concept should not require description (no description is ok).
- during REST API call; instead of having description array with only one empty description (like the below one), we should have a Json file without description array. 

```
  "descriptions": [
    {
      "uuid": "a465ad50cab944c2967e2cf763660b88",
      "external_id": null,
      "description": "",
      "locale": "en",
      "locale_preferred": true,
      "description_type": "None",
      "type": "ConceptDescription"
    }
  ],
```

**Comment by djazayeri** [comment]
I just tried to test this, and I hit several issues.

1. I try to create a concept where I left the description field blank. I get the error message that description is required. (I was expecting it to allow me to save this concept, and the server would remove the blank description that's submitted.)

2. I try to create a concept with no description (clicking the Trash icon next to the description), and I get a stack trace: http://pastebin.com/6dg1vV8M
And after that if I try to look at the concepts in my source (https://staging.openconceptlab.org/users/djazayeri/sources/OMRS/concepts/) I get stack trace: http://pastebin.com/jT6SKHDn
I think the concept was actually created, but it can't be displayed.

**Comment by djazayeri** [comment]
I tried to test this again (after new code was deployed, I think) and what I described for 1 in my previous comment still fails.
(I did not try 2 again)

**Comment by djazayeri** [comment]
I tested again after new code was deployed, and what I described above as 1 fails, but 2 now works correctly (and the json is right).

My expected behavior is that if I click on New Concept, and I (1) give a legal ID and name (2) don't touch the description section of the form at all, and (3) save
=> this should be allowed, and the web layer should just discard the blank description.

This isn't technically part of this story as-written, but can we just fix it here, and avoid creating another card?
